python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1782,6 +1785,22 @@ def my_dbt_op(dbt: DbtCliResource):
             )
 
 
+def parse_cli_vars_from_args(args: Sequence[str]) -> Dict[str, Any]:
+    var_arg_idx = next((idx for idx, arg in enumerate(args) if arg.startswith("--vars")), None)
+    if var_arg_idx is None:
+        return {}
+
+    var_arg = args[var_arg_idx]
+    if var_arg.startswith("--vars="):
+        var_string = var_arg.split("=", 1)[1]
+    elif var_arg == "--vars" and len(args) > var_arg_idx + 1:
+        var_string = args[var_arg_idx + 1]
+    else:
+        return {}
+
+    return parse_cli_vars(var_string)
```

Comment:
Could we do something like https://github.com/dagster-io/dagster/blob/6c8569f862cc6520253447d38ecb1f1ae76bac1c/python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py#L516-L519 instead

Reply:
ah... yeah.  this is exactly what I want.

I tried doing this with dbt's click cli command and found it a bit difficult.

I also looked at argparse but didn't know about `parse_known_args`.